### DISCUSSION TITLE
cpptools crash (Connection to server got closed. Server will not be restarted.)

### DISCUSSION THREAD

**Discussion by linchuan4028** [microsoft/vscode-cpptools#10322] [General]
I encountered the cpptools crash, can anyone help to find the root cause?

```
[Error - 10:19:23 AM] Connection to server got closed. Server will not be restarted.
[Error - 10:19:23 AM] Server initialization failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 10:19:23 AM] cpptools client: couldn't create connection to server.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
```

**Comment by sean-mcmanus** [comment]
Do you have a repro? What OS is it? What version of the C/C++ extension?

**Reply by linchuan4028** [reply]
OS:macOS Monterey, Version 12.6
C/C++ extension version:v1.13.8

I still encounter the issue even though I down grade to any version.

**Reply by sean-mcmanus** [reply]
It crashes for even the simplest workspace folder or open file? Did you try uninstalling/reinstalling the extension? Are you able to get a crash calls tack somehow?

**Reply by linchuan4028** [reply]
remove some folders on my project, it's work now. But I don't know the real root cause and how to get the crash call stack. Can you share the method?

**Reply by sean-mcmanus** [reply]
FYI, https://github.com/microsoft/vscode-cpptools/releases/tag/v1.14.0 has some known crash fixes.

We have a wiki with some info on how to get a crash call stack at https://github.com/microsoft/vscode-cpptools/wiki/Attaching-debugger-to-cpptools-or-cpptools%E2%80%90srv , but that only describes an approach that works if you're able to attach before it crashes. Otherwise, you'd have to explore configuring your OS to create a coredump.
